Content.Server/Dragon/DragonSystem.cs

```diff
@@ -0,0 +1,195 @@
+using Content.Server.Body.Components;
+using Content.Server.Body.Systems;
+using Content.Server.Destructible;
+using Content.Server.DoAfter;
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.CharacterAppearance.Components;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Damage;
+using Content.Shared.MobState;
+using Content.Shared.MobState.Components;
+using Content.Shared.Tag;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.Player;
+using System.Threading;
+
+namespace Content.Server.Dragon
+{
+    public sealed class DragonSystem : EntitySystem
+    {
+        [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly BloodstreamSystem _bloodstreamSystem = default!;
+        [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+        
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DragonComponent, ComponentStartup>(OnStartup);
+            SubscribeLocalEvent<DragonComponent, DevourActionEvent>(OnDevourAction);
+            SubscribeLocalEvent<DragonComponent, CarpBirthEvent>(OnCarpBirthAction);
+
+            SubscribeLocalEvent<TargetDevourSuccessfulEvent>(OnDevourSucsessful);
+            SubscribeLocalEvent<TargetDevourCancelledEvent>(OnDevourCancelled);
+            SubscribeLocalEvent<DragonComponent, MobStateChangedEvent>(OnMobStateChanged);
+        }
+
+        private void OnMobStateChanged(EntityUid uid, DragonComponent component, MobStateChangedEvent args)
+        {
+            //Empties the stomach upon death
+            //TODO: Do this when the dragon gets butchered instead
+            if (args.CurrentMobState.IsDead())
+            {
+                SoundSystem.Play(Filter.Pvs(uid), "/Audio/Animals/sound_creatures_space_dragon_roar.ogg");
+                component.DragonStomach.EmptyContainer();
+            }
+        }
+
+        /// <summary>
+        /// On cancellation of a devour attempt
+        /// </summary>
+        private void OnDevourCancelled(TargetDevourCancelledEvent args)
+        {
+            args.DragonComponent.CancelToken = null;
+        }
+
+        /// <summary>
+        /// On a sucsessful devour attempt
+        /// </summary>
+        private void OnDevourSucsessful(TargetDevourSuccessfulEvent args)
+        {
+            //TODO: Figure out a better way of removing structures via devour that still entails standing still and waiting for a DoAfter. Somehow.
+            EntityManager.QueueDeleteEntity(args.Target);
+            SoundSystem.Play(Filter.Pvs(args.User), "/Audio/Effects/sound_magic_demon_consume.ogg");
+        }
```

Comment:
this needs to be made more obvious that this is only for non-mobs, got kind of confused since the two different kind of devours arent differentiated much by name

Reply:
Also pass in entitymanager to the filter + use the proxy method for queuedeleteentity (queuedel iirc).

Reply:
I renamed the method but what proxy method and what soundsystem const?